googleapiclient/model.py

```diff
@@ -258,10 +258,8 @@ def serialize(self, body_value):
     return json.dumps(body_value)
 
   def deserialize(self, content):
-    try:
-        content = content.decode('utf-8')
-    except AttributeError:
-        pass
+    if isinstance(content, bytes):
+      content = content.decode('utf-8')
```

Comment:
I think a string should mean a textual string, basically unicode in py2 and str in py3.  And for compatibility all bytes inputs should be handled as utf-8 encoded strings.